npm_and_yarn/spec/dependabot/npm_and_yarn/update_checker/registry_finder_spec.rb

```diff
@@ -46,6 +46,18 @@
 
     it { is_expected.to eq("https://registry.npmjs.org") }
 
+    context "with no rc and with credentials" do
+      let(:credentials) do
+        [{
+          "type" => "npm_registry",
+          "registry" => "http://example.com",
+          "replaces-base" => true
+        }]
+      end
+
+      it { is_expected.to eq("http://example.com") }
```

Comment:
It makes sense to preserve the protocol but it currently will have limited use. On dotcom/ghes:
- the protocol is stripped on the backend
- the proxy won't attach credentials to http requests

Reply:
Ah, that's too bad. It will be there if we need it.